ncm-download/src/main/perl/download.pm

```diff
@@ -61,113 +61,34 @@ sub Configure
     my $defserver = $tree->{server} || "";
     my $defproto = $tree->{proto} || "";
 
-    my @proxyhosts = @{$tree->{proxyhosts} || []};
+    my $proxyhosts = $tree->{proxyhosts} || [];
 
     foreach my $esc_fn (sort keys %{$tree->{files}}) {
         my $fn = unescape($esc_fn);
         my $file = $tree->{files}->{$esc_fn};
 
-        my $source = $file->{href};
+        # Sanitize file details
+        $file->{server} = $defserver if ! exists($file->{server});
+        $file->{proto} = $defproto if ! exists($file->{proto});
 
-        my $gss;
-        if ($file->{gssapi}) {
-            $gss = $self->get_gss_token();
-            return 0 if ! $gss;
-        }
-
-        if ($source !~ m{^[a-z]+://.*}) {
-            # an incomplete URL... let's add defaults
-            if ($source !~ m{^/}) {
-                $source = "/$source";
-            }
-            if ($source !~ m{^//}) {
-                my $server = exists($file->{server}) ? $file->{server} : $defserver;
-
-                if (! $server) {
-                    # TODO: ignoring with an error?
-                    #       better would be to force this being non-empty via schema
-                    #       (eg no empty server attribute; or no empty server without default server specified),
-                    #       and not support ignoring files with error reported
-                    $self->error("$fn requested but no server, ignoring");
-                    next;
-                }
-                $source = "//$server$source";
-            }
-
-            if ($source =~ m{^//}) {
-                # server specified, but no proto
-                my $proto = exists($file->{proto}) ? $file->{proto} : $defproto;
-                if (!$proto) {
-                    # TODO: ignoring with error?
-                    $self->error("$fn requested but no proto, ignoring");
-                    next;
-                }
-                $source = "$proto:$source";
-            }
-        }
-
-        my $min_age = $file->{min_age} || 0;
+        $file->{min_age} = 0 if ! $file->{min_age};
```

Comment:
This change is not a pure refactoring - can the rationale be called out in the commit log please? (I agree it makes sense to remove the error() calls and fix them at the schema, but why not also fix the schema as part of this commit?)